xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -80,19 +80,33 @@
     @Inject
     private ConfigurationSource configuration;
 
+    private boolean threadStarted;
+
     @Override
     public void initialize()
     {
         this.queue = this.blockingQueueProvider.initBlockingQueue();
         this.consumers = new ArrayList<>();
-        int nbThreads = this.configuration.getProperty("mentions.poolSize", 1);
-        for (int i = 0; i < nbThreads; i++) {
-            startConsumer();
-        }
         JMXMentionsMBean mbean = new JMXMentions(this.queue, () -> this.consumers.size());
         this.jmxRegistration.registerMBean(mbean, MBEAN_NAME);
     }
 
+    @Override
+    public void startThreads()
+    {
+        if (!this.threadStarted) {
```

Comment:
Sounds like you forgot to remove this one

Reply:
This one is to prevent to enter the synchronized block when threadStarted is already set to true, for performance reason. Otherwise it is necessary to enter the synchronized block at every call.
But since this method is not supposed to be called very often anyway, it could be removed from simplicity's sake

Reply:
I think it's better to remove it yes.